Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -3699,6 +3699,18 @@ RefPtr<NativeImage> MediaPlayerPrivateGStreamer::nativeImageForCurrentTime()
 }
 #endif // USE(GSTREAMER_GL)
 
+RefPtr<VideoFrame> MediaPlayerPrivateGStreamer::videoFrameForCurrentTime()
+{
+    Locker sampleLocker { m_sampleMutex };
+
+    if (!GST_IS_SAMPLE(m_sample.get()))
+        return nullptr;
+
+    auto sample = adoptGRef(gst_sample_ref(m_sample.get()));
```

Comment:
I think it makes more sense to just uses the copy constructor or operator= instead convoluting it like this. Something like:

```
RefPtr<GstSample> sample = m_sample;
```